nick_work/fdaaa_port.py

```diff
@@ -0,0 +1,372 @@
+            if soup.find('primary_purpose') is not None:
+                primary_purpose = soup.find('primary_purpose').text
+            else:
+                primary_purpose = None
+
+            #is_fda_regulated
+            df_row = fda_reg_df.loc[fda_reg_df['nct_id'] == nct_id]
+            if df_row.empty:
+                is_fda_regulated = None
+            else:
+                is_fda_regulated = df_row['is_fda_regulated'].values[0]
+
+            #study_status
+            study_status = (soup.overall_status).text
+
+            #start_date
+            sd = soup.start_date
+            if sd is not None:
+                try:
+                    start_date = datetime.strptime(sd.text, '%B %d, %Y').date()
+                except ValueError:
+                    start_date = datetime.strptime(sd.text, '%B %Y').date() + relativedelta(months=+1) - timedelta(days=1)
+            else:
+                start_date = None
+
+            #available_completion_date
+            pcd = soup.primary_completion_date
+            if pcd is not None:
+                try:
+                    primary_completion_date = datetime.strptime(pcd.text, '%B %d, %Y').date()
+                    defaulted_pcd = 0
+                except ValueError:
+                    primary_completion_date = datetime.strptime(pcd.text, '%B %Y').date() + relativedelta(months=+1) - timedelta(days=1)
+                    defaulted_pcd = 1
+            else:
+                primary_completion_date = None
+                defaulted_pcd = 0
+
+            cd = soup.completion_date
+            if cd is not None:
+                try:
+                    completion_date = datetime.strptime(cd.text, '%B %d, %Y').date()
+                    defaulted_cd = 0
+                except ValueError:
+                    completion_date = datetime.strptime(cd.text, '%B %Y').date() + relativedelta(months=+1) - timedelta(days=1)
+                    defaulted_cd = 1
+            else:
+                completion_date = None
+                defaulted_cd = 0
+
+            if primary_completion_date == None and completion_date == None:
+                available_completion_date = None
+            elif primary_completion_date == None and completion_date is not None:
+                available_completion_date = completion_date
+                used_cd = 1
+                used_pcd = 0
+            else:
+                available_completion_date = primary_completion_date
+                used_pcd = 1
+                used_cd = 0
+
+            #act_flag - #looks like != handles nulls in Python so no need for extra term we use in BQ
+            if study_type == 'Interventional' and\
+            (fda_reg_drug == 'Yes' or fda_reg_device == 'Yes') and\
+            (
+             phase == 'Phase 1/Phase 2' or phase == 'Phase 2' or\
+             phase == 'Phase 2/Phase 3' or phase == 'Phase 3' or\
+             phase == 'Phase 4' or phase == 'N/A'
+            ) and\
```

Comment:
We use this expression repeatedly. Probably worth making a method like

```python
def phase_is_act_compatible(phase):
    return phase in ['Phase 1/Phase 2', 'Phase 2', 'Phase 2/Phase 3' , 'Phase 3', 'Phase 4', 'N/A']
```

Reply:
Created a whole host of functions now.